### PR TITLE
lsp: fix logging, more changes to allow calling as a library

### PR DESCRIPTION
**Pull request by landism**
#14 mostly worked, but it turned out if you ran `tilt lsp`, instead of getting usage showing you `tilt lsp start`, it just hung, because we were defining a `Run` on `lsp`. Also it broke logging even when running from this repo.

Making loglevel global is unfortunate, but:
1) cobra's support for context is pretty poor
2) cobra leans heavily into globals and trying to avoid them can sometimes really feel like swimming upstream

TLDR is: we need to create the log level when initializing the context and to update the log level in PreRun, but PreRun doesn't have access to initialize a context. There might be a way to do this without a global, but I've probably already spent too much time on that.

### CODE REVIEW

**Review by nicksieger — APPROVED** [review]
Hey, the approach ends up being less code!
